src/utils/errors.js

```diff
@@ -0,0 +1,41 @@
+import _ from 'underscore';
+import messages from './error-messages';
+
+
+class BaseError extends Error {
+    constructor(info, code) {
+        super();
+
+        if (Error.captureStackTrace) {
+            Error.captureStackTrace(this, this.constructor);
+        }
+
+        // this.code = this.code;
+        this.info = info;
+        this.code = code;
+
+        let template = _.template(messages[this.code], {
+            interpolate: /\{\{([^}]*)\}\}/g,
+            variable: 'info'
+        });
+
+        this.message = template(info);
+    }
+}
+
+class JobStartError extends BaseError {
+    constructor(info) {
+        super(info, 'JOB-START-ERROR');
+    }
+}
+
+class ViewParamsError extends BaseError {
```

Comment:
Maybe `s/View/JuttleView/g` where you are referring to juttle views, because "view" and "viewComponent" are pretty generic.